test/jdk/java/util/Currency/ValidateISO4217.java

```diff
@@ -28,285 +29,293 @@
-                            testCurrencies.add(Currency.getInstance(currency));
-                        }
-                    }
                 }
-                int index = toIndex(country);
-                testCountryCurrency(country, currency, Integer.parseInt(numeric),
-                    Integer.parseInt(minorUnit), index);
             }
         }
+        int index = toIndex(country);
+        ISO4217Codes.add(Arguments.of(country, currency, Integer.parseInt(numeric),
+                Integer.parseInt(minorUnit), index));
+        codes[index] = DEFINED;
+    }
+
+    // Generates a unique index for an alpha-2 country
+    private static int toIndex(String country) {
+        return ((country.charAt(0) - 'A') * ALPHA_NUM + country.charAt(1) - 'A');
+    }
+
+    private static void createDateFormat() {
+        format = new SimpleDateFormat("yyyy-MM-dd-HH-mm-ss", Locale.US);
+        format.setTimeZone(TimeZone.getTimeZone("GMT"));
+        format.setLenient(false);
+    }
 
-        for (int i = 0; i < additionalCodes.length; i++) {
-            int index = toIndex(additionalCodes[i][0]);
-            if (additionalCodes[i][1].length() != 0) {
-                testCountryCurrency(additionalCodes[i][0], additionalCodes[i][1],
-                    Integer.parseInt(additionalCodes[i][2]),
-                    Integer.parseInt(additionalCodes[i][3]), index);
-                testCurrencies.add(Currency.getInstance(additionalCodes[i][1]));
+    // Process 'extraCodes', turning them into JUnit arguments and populate
+    // both additionalCodes and testCurrencies.
+    private static void setUpAdditionalCodes() {
+        String[][] extraCodes = {
+                /* Defined in ISO 4217 list, but don't have code and minor unit info. */
+                {"AQ", "", "", "0"},    // Antarctica
+                /*
+                 * Defined in ISO 4217 list, but don't have code and minor unit info in
+                 * it. On the other hand, both code and minor unit are defined in
+                 * .properties file. I don't know why, though.
+                 */
+                {"GS", "GBP", "826", "2"},      // South Georgia And The South Sandwich Islands
+                /* Not defined in ISO 4217 list, but defined in .properties file. */
+                {"AX", "EUR", "978", "2"},      // \u00c5LAND ISLANDS
+                {"PS", "ILS", "376", "2"},      // Palestinian Territory, Occupied
+                /* Not defined in ISO 4217 list, but added in ISO 3166 country code list */
+                {"JE", "GBP", "826", "2"},      // Jersey
+                {"GG", "GBP", "826", "2"},      // Guernsey
+                {"IM", "GBP", "826", "2"},      // Isle of Man
+                {"BL", "EUR", "978", "2"},      // Saint Barthelemy
+                {"MF", "EUR", "978", "2"},      // Saint Martin
+                /* Defined neither in ISO 4217 nor ISO 3166 list */
+                {"XK", "EUR", "978", "2"},      // Kosovo
+        };
```

Comment:
I'd prefer this array to be placed as static in the class, not within the method.

Reply:
Fixed, as well as for `otherCodes`